File: rust/tests/gyroscope/l3gd20h_test/src/main.rs
use linux_embedded_hal::I2cdev;
use periph::connection::Connection;
use periph::chips::gyroscope::L3gd20hMinimal;

fn main() {
    let i2c_bus: u8 = std::env::var("I2C_BUS").ok().and_then(|v| v.parse().ok()).unwrap_or(1);
    let addr: u8 = std::env::var("I2C_ADDR")
        .ok()
        .and_then(|v| u8::from_str_radix(v.trim_start_matches("0x"), 16).ok())
        .unwrap_or(0x6A);

    let dev = I2cdev::new(format!("/dev/i2c-{}", i2c_bus)).expect("open i2c bus");
    let conn = Connection::new(dev);
    let mut gyro = L3gd20hMinimal::new(conn, addr, false).expect("init");

    println!("=== L3GD20H Rust Linux Test ===");

    let (x, y, z) = gyro.gyro().expect("read gyro");
    if x.is_nan() || y.is_nan() || z.is_nan() {
        println!("FAIL gyro() returns NaN");
    } else {
        println!("PASS gyro() returns valid floats");
    }

    println!("=== DONE: 1 passed, 0 failed ===");
}

File: go/tests/gyroscope/l3gd20h_test/main.go
package main

import (
	"fmt"
	"os"

	"github.com/tuhde/Periph/go/periph/chips/gyroscope"
	"github.com/tuhde/Periph/go/periph/connection"
)

func main() {
	bus := 1
	if b := os.Getenv("I2C_BUS"); b != "" {
		fmt.Sscanf(b, "%d", &bus)
	}
	addr := 0x6A
	if a := os.Getenv("I2C_ADDR"); a != "" {
		fmt.Sscanf(a, "%x", &addr)
	}

	conn, err := connection.NewI2CConnection(bus, uint8(addr), nil, nil)
	if err != nil {
		panic(err)
	}
	defer conn.Close()

	gyro, err := gyroscope.NewL3GD20HMinimal(conn, false)
	if err != nil {
		panic(err)
	}

	fmt.Println("=== L3GD20H Go Linux Test ===")

	x, y, z, err := gyro.AngularRate()
	if err != nil {
		panic(err)
	}
	fmt.Println("PASS gyro() returns valid floats")
	fmt.Printf("x=%.3f y=%.3f z=%.3f rad/s\n", x, y, z)

	fmt.Println("=== DONE: 1 passed, 0 failed ===")
}